Graphs/Table/index.js

```diff
@@ -347,27 +404,54 @@ class Table extends AbstractGraph {
             const selectedData = this.getSelectedRows()
 
             if(selectedData.length > 1) {
-                rows = selectedData
+                rows = selectedData.map( d => this.replaceKeyFromColumn(d))
             } else {
                 let row =  selectedData.length ? selectedData[0] : {}
                 /**
-                 * Compare `selectedColumn` value with all available datas and if equal to selected row,
+                 * Compare `matchingRowColumn` value with all available datas and if equal to selected row,
                  * then save all matched records in store under "matchedRows",
                 **/
-                if(selectedColumn) {
-                    const value = columnAccessor({column: selectedColumn})
-                    matchingRows = data.filter( (d) => {
-                        return (value(row) || value(row) === 0) && row !== d && value(row) === value(d)
+                if(matchingRowColumn) {
+                    let key = this.getKeyByColumnName(matchingRowColumn) || matchingRowColumn
+
+                    matchingRows = this.filterData.filter( (d) => {
+                        return (row[key] || row[key] === 0) && row !== d && row[key] === d[key]
                     });
-                }
 
-                rows = row
+                    if(matchingRows.length) {
+                        matchingRows = matchingRows.map( d => this.replaceKeyFromColumn(d))
+                    }
+                }
+                rows = this.replaceKeyFromColumn(row)
             }
             onSelect({rows, matchingRows});
         }
 
     }
 
+    // replace random generated keys of columns from actual column name
+    replaceKeyFromColumn(row) {
+        const columns = this.getKeyColumns()
+        let updatedRow = {}
+        for (let key in row) {
+            if(row.hasOwnProperty(key)) {
+                objectPath.set(updatedRow, columns[key] ? columns[key].column : key, row[key]);
+            }
+        }
+        return updatedRow
+    }
+
+    // pass column name and get column key (random generated at component will mount/ update)
+    getKeyByColumnName(column) {
+        const columns = this.getKeyColumns()
+        for (let key in columns) {
```

Comment:
can we use array filter here?

Reply:
Here `columns` is an object, so we can't use array filter.